### DISCUSSION TITLE
Struggling with Todo List Tutorial

### DISCUSSION THREAD

**Discussion by kuraiji** [AvaloniaUI/Avalonia#8159] [Q&A] [answered]
At the "Creating a View/ Wiring Up the Views" step, I can't get Content="{Binding List}" to work. I get the error "cannot resolve symbol "List" due to unknown DataContext". 

I noticed that bring back the Design.DataContext element...
`<Design.DataContext>
        <vm:MainWindowViewModel/>
    </Design.DataContext>`
...makes the symbol resolve. But doing that brings in a new error. The MainWindowViewModel class doesn't have a default constructor, so it breaks it since <vm:MainWindowViewModel/> calls for a default constructor.

I have no clue what I should be doing. The tutorial doesn't have the design.datacontext element at all. My project is located [here](https://github.com/kuraiji/WalkmanPlaylistCreatorGUI/tree/master/Speedy%20Walkman%20Playlist%20Creator%20GUI).

**Comment by timunie** [comment, marked as answer]
You can also specify x:DataType on your Window or on any XAML node. 

I recommend to use CompiledBindings anyway. You can read more about it in the Docs here https://docs.avaloniaui.net/docs/data-binding/compiledbindings

Happy coding
Tim

**Comment by kuraiji** [comment]
For anyone looking at this in the future: Adding this after x:class in the MainWindow.axaml did the trick
`x:DataType="vm:MainWindowViewModel"`
`x:CompileBindings="True"`
